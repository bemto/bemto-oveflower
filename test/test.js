import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import Overflower from '..';

const testSnapshot = function(tag, props, children) {
  const tree = renderer.create(
    React.createElement(
      tag,
      props,
      children
    )
  ).toJSON();

  expect(tree).toMatchSnapshot();
}

test('Just an overflower', () => {
  testSnapshot(
    Overflower,
    {
      __Overflow: 'Short text'
    },
    'Some very long text'
  );
});

test('Just an overflower without overflow', () => {
  testSnapshot(
    Overflower,
    {},
    'Some very long text'
  );
});
