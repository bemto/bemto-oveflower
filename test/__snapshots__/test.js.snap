// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Just an overflower 1`] = `
.c0 {
  display: inline-block;
  overflow: hidden;
  box-sizing: border-box;
  max-width: 100%;
  height: 1.5em;
  line-height: 1.5em;
  white-space: nowrap;
  text-overflow: ellipsis;
}

.c0__Original {
  display: inline;
}

.c0_Overflow {
  display: none;
}

@supports (-webkit-flex-wrap:wrap) or (-ms-flex-wrap:wrap) or (flex-wrap:wrap) {
  .c0 {
    display: -webkit-inline-box;
    display: -webkit-inline-flex;
    display: -ms-inline-flexbox;
    display: inline-flex;
    -webkit-flex-wrap: wrap;
    -ms-flex-wrap: wrap;
    flex-wrap: wrap;
  }

  .c0__Original {
    -webkit-flex-basis: 100%;
    -ms-flex-preferred-size: 100%;
    flex-basis: 100%;
  }

  .c0__Original:first-child {
    overflow: hidden;
    text-overflow: ellipsis;
  }

  .c0__Overflow {
    display: block;
    overflow: hidden;
    -webkit-box-flex: 1;
    -webkit-flex-grow: 1;
    -ms-flex-positive: 1;
    flex-grow: 1;
    width: 0;
    text-overflow: ellipsis;
  }
}

<span
  className="c0"
>
  <span
    aria-hidden={true}
    className="__Overflow c0__Overflow"
    title="Some very long text"
  >
    Short text
  </span>
  <span
    className="__Original c0__Original"
  >
    Some very long text
  </span>
</span>
`;

exports[`Just an overflower without overflow 1`] = `
.c0 {
  display: inline-block;
  overflow: hidden;
  box-sizing: border-box;
  max-width: 100%;
  height: 1.5em;
  line-height: 1.5em;
  white-space: nowrap;
  text-overflow: ellipsis;
}

.c0__Original {
  display: inline;
}

.c0_Overflow {
  display: none;
}

@supports (-webkit-flex-wrap:wrap) or (-ms-flex-wrap:wrap) or (flex-wrap:wrap) {
  .c0 {
    display: -webkit-inline-box;
    display: -webkit-inline-flex;
    display: -ms-inline-flexbox;
    display: inline-flex;
    -webkit-flex-wrap: wrap;
    -ms-flex-wrap: wrap;
    flex-wrap: wrap;
  }

  .c0__Original {
    -webkit-flex-basis: 100%;
    -ms-flex-preferred-size: 100%;
    flex-basis: 100%;
  }

  .c0__Original:first-child {
    overflow: hidden;
    text-overflow: ellipsis;
  }

  .c0__Overflow {
    display: block;
    overflow: hidden;
    -webkit-box-flex: 1;
    -webkit-flex-grow: 1;
    -ms-flex-positive: 1;
    flex-grow: 1;
    width: 0;
    text-overflow: ellipsis;
  }
}

<span
  className="c0"
>
  <span
    className="__Original c0__Original"
  >
    Some very long text
  </span>
</span>
`;
